src/components/transactions/utils/categories/CategoriesItems.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { categoryIcons } from '../../../../utils/constants/icons';
+import { InputAdornment, MenuItem, styled } from '@mui/material';
+import { ReactComponent as AdjustmentIcon } from '../../../../assets/icons/shared/adjustment.svg';
+import { useTranslation } from 'react-i18next';
+import {
+  AddButtonSvg,
+  CancelSearchSvg,
+  SearchField,
+  SelectHeader,
+  SelectHeaderButton,
+  TextInputField,
+} from '../../../../theme/global';
+import { ReactComponent as SearchIcon } from '../../../../assets/icons/shared/search.svg';
+import { ReactComponent as CancelSearchIcon } from '../../../../assets/icons/shared/cancelSearch.svg';
+import { ReactComponent as PlusIcon } from '../../../../assets/icons/shared/plus.svg';
+
+const CategoriesMenuItem = styled(MenuItem)((props) => ({
+  borderRadius: props.theme.borderRadius,
+  backgroundColor: props.theme.colors.background.primary,
+}));
+
+const InfoContainer = styled('div')((props) => ({
+  display: 'flex',
+  alignItems: 'center',
+  color: props.theme.colors.text.primary,
+}));
+
+const CategoriesItemSvg = styled('svg')((props) => ({
+  width: 34,
+  height: 34,
+  marginRight: props.theme.spacing(3),
+}));
+
+function CategoriesItems({
+  categories,
+  category,
+  setCategory,
+  setOpenCategoryDialog,
+}) {
+  const { t } = useTranslation();
+
+  return (
+    <TextInputField
+      margin="normal"
+      required
+      select
+      label={t('INFO_TRANSACTION.CATEGORY')}
+      value={category}
+      onChange={(event) => setCategory(event.target.value)}
+    >
+      <SelectHeader>
+        {t('INFO_TRANSACTION.AVAILABLE_CATEGORIES')}
+        <SelectHeaderButton>
+          <AddButtonSvg
+            onClick={() => setOpenCategoryDialog(true)}
+            as={PlusIcon}
+          />
+        </SelectHeaderButton>
+      </SelectHeader>
+      <SearchField
+        placeholder={t('INFO_TRANSACTION.SEARCH_CATEGORY')}
+        InputProps={{
+          startAdornment: (
+            <InputAdornment position="start">
+              <SearchIcon />
+            </InputAdornment>
+          ),
+          endAdornment: (
+            <InputAdornment position="end">
+              <CancelSearchSvg as={CancelSearchIcon} />
+            </InputAdornment>
+          ),
+        }}
+      />
+      {categories.length === 0 ? (
+        <div>{t('INFO_TRANSACTION.NO_CATEGORIES')}</div>
+      ) : (
+        categories.map((category, index) => {
+          let Icon = AdjustmentIcon;
```

Comment:
Rather than `let`/`if`, it would be better to use a ternary operator here like:

```
const Icon = category.description === 'Balance adjustment'
  ? AdjustmentIcon
  : categoryIcons[category.icon];
```